cmd/kaniko-docker/main.go

```diff
@@ -475,7 +496,16 @@ func createDockerCfgFile(username, password, registry string) error {
 
 	authBytes := []byte(fmt.Sprintf("%s:%s", username, password))
 	encodedString := base64.StdEncoding.EncodeToString(authBytes)
-	jsonBytes := []byte(fmt.Sprintf(`{"auths": {"%s": {"auth": "%s"}}}`, registry, encodedString))
+	var jsonBytes []byte
+
+	if dockerRegistry != "" {
+		authBytesBase := []byte(fmt.Sprintf("%s:%s", dockerUsername, dockerPassword))
+		encodedStringBase := base64.StdEncoding.EncodeToString(authBytesBase)
+		jsonBytes = []byte(fmt.Sprintf(`{"auths": {"%s": {"auth": "%s"}}, {"%s": {"auth": "%s"}}}`,
+			registry, encodedString, dockerRegistry, encodedStringBase))
+	} else {
+		jsonBytes = []byte(fmt.Sprintf(`{"auths": {"%s": {"auth": "%s"}}}`, registry, encodedString))
+	}
 
 	if err := writeDockerCfgFile(jsonBytes); err != nil {
 		return errors.Wrap(err, "failed to write docker config file")
```

Comment:
Let's use `docker.CreateDockerCfgFile` utility (or create different method) for this. This can be used by others (gar, acr) as well.